project-addons/product_stock_unsafety/product.py

```diff
@@ -72,7 +73,10 @@ def calc_joking_index(self):
             joking_tot += stock_product_id.joking
         avg = joking_tot / cont
         for product in product_obj.search([]):
-            if product.type != 'product' or product.id not in product_ids:
+            product_tmpl = self.env['product.template'].browse(int(product.product_tmpl_id))
+            brand = self.env['product.brand'].browse(int(product_tmpl.product_brand_id))
+            if product.type != 'product' or product.id not in product_ids \
+                    or brand.not_compute_joking:
```

Comment:
¿No funcionaba como te puse yo? en principio debería y te ahorrabas instanciar product_tmpl y brand sería:
if product.type != 'product' or product.id not in product_ids or (product.product_brand_id and product.product_brand_id.id or False):